src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1678,10 +1678,13 @@ private void CheckModelAndSyntaxNodeToSpeculate(CSharpSyntaxNode syntax)
                 lookupResult.Free();
             }
 
-            ImmutableArray<ISymbol> sealedResults = results.ToImmutableAndFree();
-            return name == null
-                ? FilterNotReferencable(sealedResults)
-                : sealedResults;
+            var result = name == null
+                ? FilterNotReferenceable(results)
+                : results.ToImmutable();
+
+            results.Free();
```

Comment:
This can be improved by filtering in place:

```cs
if (name == null)
    FilterNotReferenceable(results);

var result = results.ToImmutableAndFree();
```

Reply:
Let me give that a whirl, I think it's going to complicate FilterNotReferenceable a bit, but hopefully not too much

Reply:
Yep, I like that better.